src/Zicht/Bundle/FrameworkExtraBundle/Command/CreateStdLoggerTrait.php

```diff
@@ -0,0 +1,64 @@
+<?php
+/**
+ * @copyright Zicht Online <https://zicht.nl>
+ */
+
+namespace Zicht\Bundle\FrameworkExtraBundle\Command;
+
+use Monolog\Handler\FilterHandler;
+use Monolog\Handler\StreamHandler;
+use Monolog\Logger;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+trait CreateStdLoggerTrait
+{
+    /**
+     * Returns a logger configured to use STDOUT and STDERR depending on $input and $output settings
+     *
+     * Use this method to construct a logger that sends log messages to stdout or stderr.  This
+     * is especially useful for commands that are executed using crontab in combination with the
+     * linux tool cronic.
+     *
+     * Usage:
+     * ```php
+     * class ExampleCommand extends AbstractCommand
+     * {
+     *     use CreateStdLoggerTrait;
+     *
+     *     protected function execute(InputInterface $input, OutputInterface $output)
+     *     {
+     *         $logger = $this->createStdLogger($output, 'console');
+     *         $logger->debug(...);      // logs to stdout when -v or --verbose
+     *         $logger->info(...);       // logs to stdout
+     *         $logger->notice(...);     // logs to stdout
+     *         $logger->warning(...);    // logs to stdout
+     *         $logger->error(...);      // logs to stderr
+     *         $logger->critical(...);   // logs to stderr even when -q (quiet)
+     *         $logger->alert(...);      // logs to stderr even when -q (quiet)
+     *         $logger->emergency(...);  // logs to stderr even when -q (quiet)
+     *     }
+     * }
+     * ```
+     *
+     * @param OutputInterface $output
+     * @param string $name
+     * @return LoggerInterface
+     */
+    protected function createStdLogger(OutputInterface $output, string $name): LoggerInterface
+    {
+        $levelMap = [
+            OutputInterface::VERBOSITY_QUIET => Logger::CRITICAL,
+            OutputInterface::VERBOSITY_NORMAL => Logger::INFO,
+            OutputInterface::VERBOSITY_VERBOSE => Logger::DEBUG,
+            OutputInterface::VERBOSITY_VERY_VERBOSE => Logger::DEBUG,
+            OutputInterface::VERBOSITY_DEBUG => Logger::API,
+        ];
+        $level = $levelMap[$output->getVerbosity()] ?? Logger::ERROR;
+
+        $logger = new Logger($name);
+        $logger->pushHandler(new FilterHandler(new StreamHandler(STDOUT, $level), Logger::DEBUG, Logger::ERROR - 1));
```

Comment:
`Logger::ERROR - 1` Komt uit op het getal 399, maar dat is geen gangbaar error-niveau, moet dat dan niet Logger::WARNING worden, het eerste niveau onder ERROR?

Reply:
Klopt, warning is 300 en error is 400.  Maar het is mogelijk om een log message te genereren met een log level van elk willekeurig 'level'.  Dus een log van level 350 is prima mogelijk, daarom gebruik ik ERROR - 1, zodat deze ook zichtbaar is (anders wordt het namelijk zomaar verborgen, en dat lijkt me heel onwenselijk.

Dezelfde gedachtegang was ook de reden om `OutputInterface::VERBOSITY_DEBUG => Logger::LOG,` te doen bij je vorige opmerking, alleen dan voor mensen die handmatig een log van level 50 oid hadden gemaakt.  Maar ik kan me voorstellen dat dat je die dan nog steeds niet wilt hebben als de console op debug output staat.

Reply:
Lijkt me onwaarschijnlijk, maar kan evt voorkomen in onze eigen bundles waar we afwijken van de standaard.